iterators.tex

```diff
@@ -623,31 +650,186 @@
 \textit{p}).
 \exitexample
 
-\indexlibrary{\idxcode{Iterator}}%
-\begin{codeblock}
+{\color{newclr}
+\rSec2[iterators.sizediterator]{Concept SizedIterator}
+\pnum
+The \tcode{SizedIterator} concept refines \tcode{Iterator} with the
+capability to determine the distance between two iterators that denote
+elements of the same range in constant time with the \tcode{-} operator.
+
+\indexlibrary{\idxcode{SizedIterator}}%
+\begin{itemdecl}
+  template <class>
+  constexpr bool disable_sized = false;
```

Comment:
`SizedIterator` is distinct for two reasons: (1) `disable_sized`, which obviously is not significant if `disable_sized` becomes `disable_sized_sentinel`, and (2) it makes it slightly more intuitive that functions accepting a `SizedIterator` pair have different preconditions than functions that accept a `SizedSentinel` pair, i.e., "`[i,s)` shall denote a range" vs. "either `[i,j)` or `[j,i)` shall denote a range." I also think the presentation is clearer when iterator requirements are named `FooIterator`; having `RandomAccessIterator` require `SizedSentinel` will throw people off, just as you commented on the issue.

What's the use case for `disable_sized_sentinel`? A user who read the requirements, implemented an improperly constrained `-` anyway for his iterator with any random sentinel, and finds it easier to specialize `disable_sized_sentinel<I, S>` for each problem `S` as it comes up rather than fix his `-`?